homeassistant/components/climate/mysensors.py

```diff
@@ -0,0 +1,238 @@
+# pylint: disable=too-many-instance-attributes
+class MySensorsHVAC(mysensors.MySensorsDeviceEntity, ClimateDevice):
+    """Representation of a MySensorsHVAC hvac."""
+
+    def __init__(self, *args):
+        """Setup instance attributes."""
+        mysensors.MySensorsDeviceEntity.__init__(self, *args)
+        self._state = None
+        # Default Target Temperature set to a comfortable value of 24 degrees
+        self._target_temperature = 24.0
+        self._target_humidity = None
+        self._away = None
+        self._current_temperature = None
+        self._current_humidity = None
+        self._current_fan_mode = "Auto"
+        self._current_operation = "Off"
+        self._aux = None
+        self._current_swing_mode = "Off"
+        self._fan_list = ["Auto", "Min", "Normal", "Max"]
+        self._operation_list = ["CoolOn", "Off"]
+        self._swing_list = ["On", "Off"]
+
+    @property
+    def assumed_state(self):
+        """Return True if unable to access real state of entity."""
+        return self.gateway.optimistic
+
+    @property
+    def should_poll(self):
+        """Polling not needed."""
+        return False
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return (TEMP_CELSIUS
+                if self.gateway.metric else TEMP_FAHRENHEIT)
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def current_humidity(self):
+        """Return the current humidity."""
+        return self._current_humidity
+
+    @property
+    def target_humidity(self):
+        """Return the humidity we try to reach."""
+        return self._target_humidity
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        return self._current_operation
+
+    @property
+    def operation_list(self):
+        """List of available operation modes."""
+        return self._operation_list
+
+    @property
+    def is_away_mode_on(self):
+        """Return if away mode is on."""
+        return self._away
+
+    @property
+    def is_aux_heat_on(self):
+        """Return true if away mode is on."""
+        return self._aux
+
+    @property
+    def current_fan_mode(self):
+        """Return the fan setting."""
+        return self._current_fan_mode
+
+    @property
+    def fan_list(self):
+        """List of available fan modes."""
+        return self._fan_list
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        set_req = self.gateway.const.SetReq
+        if kwargs.get(ATTR_TEMPERATURE) is not None:
```

Comment:
Fixed it now... was a silly mistake... it works and is able to operate flawlessely from HA...